overview/default.nix

```
@@ -7,28 +8,32 @@
   inherit
     (builtins)
     any
+    attrNames
     attrValues
     concatStringsSep
     filter
     isList
-    mapAttrs
-    match
     readFile
+    stringLength
     substring
     toJSON
     toString
     ;
 
   inherit
-    (pkgs.lib)
+    (lib)
     concatLines
+    flattenAttrsDot
+    flip
     mapAttrsToList
     optionalString
-    unique
     ;
 
   empty = xs: assert isList xs; xs == [];
   heading = i: text: "<h${toString i}>${text}</h${toString i}>";
+  isPrefix = prefix: candidate: prefix == substring 0 (stringLength prefix) candidate;
+  concatStringsDot = concatStringsSep ".";
```

```suggestion
  join = concatStringsSep;
```

Right?

This alias should actually go to Nixpkgs...

---

I don't understand your suggestion. It removes a definition for `concatStringsDot` and adds a definition for `join` to alias `concatStringsSep`. But `concatStringsDot` was a partial application of `concatStringsSep`, not a plain alias.

I have removed `concatStringsDot` in 70146c7 because it was only used in one place anyway.

Regarding adding `join = concatStringsSep` to Nixpkgs, I do sympathize, but I am not sure I see the justification given that `concatStringsSep` is already there and adding another name adds complexity. Also, `concatStringsSep` is explicit about what is being concatenated here, while `join` is so generic that one might expect it to do other things, like operate on e.g. lists.

---

One would add it to `lib.strings`, yes, but it'd still be consumed as `lib.join`, see e.g. https://github.com/NixOS/nixpkgs/pull/254813#discussion_r1367015546